test/Ansatz/Chain_test.jl

```
@@ -51,6 +51,24 @@
         @test rightsite(qtn, Site(1)) == Site(2)
     end
 
+    @testset "truncate" begin
+        qtn = Chain(State(), Open(), [rand(2, 2), rand(2, 2, 2), rand(2, 2)])
+        canonize_site!(qtn, Site(2); direction = :right, method = :svd)
+
+        @test_throws ArgumentError truncate!(qtn, [Site(1), Site(2)]; maxdim = 1)
```

Why should this fail?

---

There are no singular values there to truncate

---

Ah ok. Maybe then `ArgumentError` is not the exception we should throw.

---

Do you have any suggestion? I can't seem to think of a better one

---

I think we should throw a custom exception of our own. `NonCanonicalFormException`? `MissingSchmidtCoefficientsException`?

---

Okey, done